youtube_dl/YoutubeDL.py

```diff
@@ -633,6 +635,10 @@ def extract_info(self, url, download=True, ie_key=None, extra_info={},
         extra_info is a dict containing the extra values to add to each result
         '''
 
+        if not ie_key and self._force_generic_extractor_required:
+            self._force_generic_extractor_required = False
```

Comment:
I think that this line would produce unexpected results with multiple urls. The first one would use the generic extractor and the rest would use the normal extractors. (I may have misunderstood it)